### ISSUE TITLE
`no-for-loop` produces invalid autofix when mutation is present

### ISSUE THREAD

**Issue by bmish** [sindresorhus/eslint-plugin-unicorn#742]
If the current element gets mutated in the loop body, we can't use `const` for it, and probably want to use `let` instead.

Original code:

```js
for (let i = 0; i < plugins.length; i++) {
  let plugin = plugins[i];
  plugin = calculateSomeNewValue();
  // ...
}
```

After autofix (invalid, fails the [no-const-assign](https://eslint.org/docs/rules/no-const-assign) lint rule):

```js
for (const plugin of plugins) {
  plugin = calculateSomeNewValue();
  // ...
}
```

Here is one autofix that would be valid:

```js
for (let plugin of plugins) {
  plugin = calculateSomeNewValue();
  // ...
}
```

**Comment by fisker** [comment]
I remember we respect the declare kind, need confirm.

**Comment by fisker** [comment]
Really sorry, that I forgot this issue, should be able to fix by #878
